internal/receiver/awscontainerinsightreceiver/cadvisor/extractors/extractor.go

```diff
@@ -0,0 +1,102 @@
+package extractors
+
+import (
+	"log"
+	"time"
+
+	aws "github.com/aws-observability/aws-otel-collector/internal/aws"
+	common "github.com/aws-observability/aws-otel-collector/internal/aws/containerinsightcommon"
+	cinfo "github.com/google/cadvisor/info/v1"
+)
+
+const (
+	containerNameLable = "io.kubernetes.container.name"
+	// TODO: https://github.com/containerd/cri/issues/922#issuecomment-423729537 the container name can be empty on containerd
+	infraContainerName = "POD"
+	Metrics            = "Metrics"
+	Dimensions         = "Dimensions"
+)
+
+func GetStats(info *cinfo.ContainerInfo) *cinfo.ContainerStats {
+	if len(info.Stats) == 0 {
+		return nil
+	}
+	// When there is more than one stats point, always use the last one
+	return info.Stats[len(info.Stats)-1]
+}
+
+type MachineInfoProvider interface {
+	GetNumCores() int
+	GetMemoryCapacity() uint64
+}
+
+type MetricExtractor interface {
+	HasValue(*cinfo.ContainerInfo) bool
+	GetValue(*cinfo.ContainerInfo, MachineInfoProvider, string) []*CAdvisorMetric
+}
+
+type CAdvisorMetric struct {
+	fields map[string]interface{}
+	tags   map[string]string
```

Comment:
Pls help to add comments how we define these 2 variables?

Reply:
sure